build.go

```diff
@@ -351,14 +384,40 @@ func errLater(err error) fileOpener {
 	}
 }
 
-// bin returns a fileOpener for either the "tailscale" or "tailscaled"
-// baseProg binary, building it or downloading it from pkgs.tailscale.com as
-// necessary.
-func bin(baseProg string, param spkParams) fileOpener {
-	if param.srcDir == "" {
-		return errLater(errors.New("TODO: download binaries from pkgs.tailscale.com"))
+// This is the same as the one github.com/tailscale/mkctr, share somehow?
+func compileGoBinary(dir, gopath string, env []string, ldflags, gotags string) (string, error) {
+	f, err := os.CreateTemp("", "out")
```

Comment:
when are these cleaned up?

Reply:
Good catch. Added docstring to say that the caller needs to clean it up and made the caller clean it up.